lib/generate_speaker_list/generator.rb

```diff
@@ -0,0 +1,52 @@
+require 'json'
+
+module GenerateSpeakerList
+  class Generator
+    SPEAKERS_DATA_FILE = 'data/speakers.json'
+
+    def initialize
+      @files = Dir['source/meetings/*/*/index.html.md'].sort.reverse
+      @talks = []
+      @authors = {}
+    end
+
+    def call
+      load_all_talks_from_files!
+
+      @talks = @talks.flatten.select { |talk| talk[:author] != nil }.group_by { |talk| talk[:author] }
+
+      @talks.each do |author_name, talks|
+        @authors[author_name] = {
+          name: author_name,
+          links: talks.collect { |talk| talk[:author_link] }.uniq.compact,
+          talks: talks.collect do |talk|
+            {
+              title: talk[:title],
+              coverage: talk[:coverage],
+              summary: talk[:summary],
+              year: talk[:year],
+              month: talk[:month]
+            }
+          end
+        }
+      end
+
+
+      pp @authors
+
+      save_authors_to_json!
+    end
+
+    private
+
+    def load_all_talks_from_files!
+      @files.each do |file|
+        @talks << ParseFile.new(file).call
+      end
+    end
+
+    def save_authors_to_json!
+      File.write(SPEAKERS_DATA_FILE, JSON.pretty_generate(@authors))
```

Comment:
Be nice if this file was sorted by name so it's easier to look through and easier to edit as we add things for future meetings.  Thinking particularly of the case where we have to add a new talk to an existing speaker.  The current order is weird so would get out of hand quickly as we added to it.